app/src/main/java/com/es/findsoccerplayers/adapter/MessageAdapter.java

```diff
@@ -0,0 +1,80 @@
+package com.es.findsoccerplayers.adapter;
+
+import android.content.Context;
+import android.text.format.DateUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.es.findsoccerplayers.models.Message;
+import com.es.findsoccerplayers.R;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.text.DateFormat;
+import java.util.Date;
+import java.util.List;
+
+public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.ViewHolder> {
+
+    private Context context;
+    private List<Message> chats;
+    private static int MSG_TYPE_LEFT = 0;
+    private static int MSG_TYPE_RIGHT = 1;
+
+    public MessageAdapter(Context c, List<Message> chats){
+        context = c;
+        this.chats = chats;
+    }
+
+    @Override
+    public MessageAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view;
+        if(viewType == MSG_TYPE_LEFT)
+            view = LayoutInflater.from(context).inflate(R.layout.frag_chat_message_left, parent, false);
+        else
+            view = LayoutInflater.from(context).inflate(R.layout.frag_chat_message_right, parent, false);
+        return new MessageAdapter.ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(MessageAdapter.ViewHolder holder, int position) {
+        Message m = chats.get(position);
+        holder.message.setText(m.getText());
+
+        String dateStr = DateUtils.formatDateTime(context, m.getTimestamp(), DateUtils.FORMAT_ABBREV_MONTH |
+                DateUtils.FORMAT_SHOW_DATE | DateUtils.FORMAT_SHOW_TIME);
```

Comment:
l'ora sarebbe meglio se mostra in formato 24H

Reply:
E' già nel formato 24h